recipes/src/main/resources/quarkus-updates/core/3.24.CR1.yaml

```diff
@@ -0,0 +1,138 @@
+  - org.openrewrite.java.ChangeMethodName:
+      methodPattern: org.hibernate.Session load(Class, Object, org.hibernate.LockMode)
+      newMethodName: get
+  - org.openrewrite.java.ChangeMethodName:
+      methodPattern: org.hibernate.Session load(String, Object, org.hibernate.LockMode)
+      newMethodName: get
+  - org.openrewrite.java.ChangeMethodName:
+      methodPattern: org.hibernate.Session load(Class, Object, org.hibernate.LockOptions)
+      newMethodName: get
+  - org.openrewrite.java.ChangeMethodName:
+      methodPattern: org.hibernate.Session load(String, Object, org.hibernate.LockOptions)
+      newMethodName: get
+  # These are deprecated but not removed. Still, they're equivalent, so applications should switch to them.
+  - org.openrewrite.java.ChangeMethodName:
+      methodPattern: org.hibernate.Session get(Class, Object)
+      newMethodName: find
+  - org.openrewrite.java.ChangeMethodName:
+      methodPattern: org.hibernate.Session get(String, Object)
+      newMethodName: find
+  # No way to handle these deprecated methods automatically.
+  # TODO: insert a TODO comment? Or just leave the deprecated calls and have people rely on compiler warnings?
+  # org.hibernate.Session get(Class, Object, org.hibernate.LockMode)
+  # org.hibernate.Session get(String, Object, org.hibernate.LockMode)
+  # org.hibernate.Session get(Class, Object, org.hibernate.LockOptions)
+  # org.hibernate.Session get(String, Object, org.hibernate.LockOptions)
+  - org.openrewrite.java.ChangeMethodName:
+      methodPattern: org.hibernate.Session delete(Object)
+      newMethodName: remove
+  # This would require removing the first argument, and I don't know how to do that.
+  # It's very unlikely to be used anyway.
+  # org.hibernate.Session delete(String, Object) => remove(Object)
+  # org.hibernate.Session refresh(String, Object) => refresh(Object)
+  # org.hibernate.Session refresh(String, Object, org.hibernate.LockOptions) => refresh(Object, org.hibernate.LockOptions)
+  - org.openrewrite.java.ChangeMethodName:
+      methodPattern: org.hibernate.Session save(Object)
+      newMethodName: persist
+  - org.openrewrite.java.ChangeMethodName:
+      methodPattern: org.hibernate.Session save(String, Object)
+      newMethodName: persist
+  # This is quite bold, as `update` can behave differently from merge, but there is no alternative.
+  - org.openrewrite.java.ChangeMethodName:
+      methodPattern: org.hibernate.Session update(Object)
+      newMethodName: merge
+  # This would map to either persist or merge depending on more context knowledge than we can possibly have at build time.
+  # We could consider inserting "if (session.isDetached(entity)) { return session.merge(entity); } else { session.persist(entity); return entity; }",
+  # but that goes against the spirit of removing this method: having to do this is a code smell to begin with.
+  # org.hibernate.Session saveOrUpdate(Object)
+  # org.hibernate.Session saveOrUpdate(String, Object)
```

Comment:
FYI @maxandersen , these are probably the most important remaining problems when it comes to `Session` -- I expect everything else is either handled by the recipe, or business-as-usual.